### PR TITLE
fix: bgcolor settings for root node

### PR DESCRIPTION
**Pull request by Lazerbeak12345**
Just noticed a todo that looked easy enough lol.

Not sold on `bg_fullscreen` as a name. I think it's less ambiguous than just `fullscreen`, but not sure what it should be.

---

semantics node: `fix:` might not be appropriate here. Arguably could be considered a feature, since new functionality is being added. Either way doesn't really matter I guess.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -459,13 +459,16 @@ local function render_ast(node)
     if node.no_prepend then
         res[#res + 1] = {type = "no_prepend"}
     end
-    if node.fbgcolor then
-        -- TODO: Fix this
+    if node.fbgcolor or node.bgcolor or node.bg_fullscreen then
         res[#res + 1] = {
             type = "bgcolor",
-            bgcolor = node.fbgcolor,
-            fullscreen = true
```
Comment (Lazerbeak12345) [inline init.lua:467]: This is a breaking change. `fbgcolor ~= bgcolor`

Comment (Lazerbeak12345) [reply init.lua:467]: minetest defaults to `bgcolor.fullscreen = false`. This is also breaking.

Comment (Lazerbeak12345) [reply init.lua:467]: (also should be noted that neither were documented features in flow, so it might not matter anyway)

**Comment by luk3yx** [comment]
Thanks for your PR. I would have liked to somehow simplify the bgcolor options but `bgcolor` could also be in the formspec prepend and need to be partially overridden. "Fix" is probably appropriate since it didn't work properly before.

**Comment by Lazerbeak12345** [comment]
Should I send a PR to document this feature?

**Comment by luk3yx** [comment]
> Should I send a PR to document this feature?

If you want to, if not I'll probably document it at some point.

**Pull request merged**